### PR TITLE
Adaptive Radau

### PR DESCRIPTION
**Pull request by d-netto**

### CODE REVIEW

File: test/algconvergence/ode_firk_tests.jl [hunk 1]
```diff
@@ -56,10 +56,6 @@ for iip in (true, false)
     vanstiff = ODEProblem{false}((u,p,t)->van(u,p,t), [0;sqrt(3)], (0.0,1.0), 1e6)
   end
   sol = solve(vanstiff, RadauIIA3())
-  if iip
-    @test sol.destats.naccept + sol.destats.nreject > sol.destats.njacs # J reuse
-    @test sol.destats.njacs < sol.destats.nw # W reuse
-  end
```
Comment (YingboMa) [inline test/algconvergence/ode_firk_tests.jl]: Oh yes, leave the Jacobian and W reuse for now. I have to reform it anyway.

File: src/perform_step/firk_perform_step.jl [hunk 2]
```diff
@@ -305,181 +323,38 @@ end
...
   end
-  cache.ηold = η
-  cache.iter = iter
 
   @. u = uprev + z2
-  f(fsallast, u, p, t+dt)
-  return
-end
-
-@muladd function perform_step!(integrator, cache::RadauIIA5ConstantCache, repeat_step=false)
```
Comment (ChrisRackauckas) [inline src/perform_step/firk_perform_step.jl:316]: why is this deleted?

Comment (d-netto) [reply src/perform_step/firk_perform_step.jl]: deleted it by accident, just fixed in the the next commits

**Comment by d-netto** [comment]
@YingboMa: the error "LoadError: type RadauIIA3Cache has no field nlsolver" persisted even after adding RadauIIA3 to the controller.jl file. Any idea of what could be causing it?

**Comment by YingboMa** [comment]
Always copy and paste the entire stacktrace. I cannot help if you just tell me the error message.

**Comment by d-netto** [comment]
> Always copy and paste the entire stacktrace. I cannot help if you just tell me the error message.

**Comment by d-netto** [comment]
Ok. The stack trace has a few thousand lines of code, unfortunately.

**Comment by d-netto** [comment]
@ChrisRackauckas, @YingboMa: stack trace above.

**Comment by ChrisRackauckas** [comment]
Set `dt=0.1` for now. It has to do with how it's trying to grab a temporary.
